cmd/minishift/cmd/util/oc.go

```diff
@@ -143,3 +146,29 @@ func GetOcPathForProfile(profileName string) (error, string) {
 	}
 	return nil, instanceCfg.OcPath
 }
+
+// PullOpenshiftImageAndCopyOcBinary pull the openshift image if not available and then copy the client binary to Remote/VM machine
+// Any occurring error is also returned.
+func PullOpenshiftImageAndCopyOcBinary(dockerCommander docker.DockerCommander, requestedOpenShiftVersion string) error {
+	fmt.Printf("-- Pulling the Openshift Container Image ")
+	progressDots := progressdots.New()
+	progressDots.Start()
+	// We need to make sure if images are already exist from the cache then don't pull it again.
+	imageExist, err := dockerCommander.IsImageExist(minishiftConstants.GetOpenshiftImageName(requestedOpenShiftVersion))
+	if !imageExist {
+		_, err = dockerCommander.Pull(minishiftConstants.GetOpenshiftImageName(requestedOpenShiftVersion))
+		if err != nil {
+			return fmt.Errorf("Error copying the oc binary to %s: %v", minishiftConstants.OcPathInsideVM, err)
```

Comment:
this step does not perform the copy... please address the error for pulling the image.